engine/base/kafka/src/main/scala/pl/touk/nussknacker/engine/baseengine/kafka/NuKafkaEngineApp.scala

```diff
@@ -28,9 +28,14 @@ object NuKafkaEngineApp extends App with LazyLogging {
 
   val scenarioInterpreter = prepareScenarioInterpreter
 
-  Using.resource(scenarioInterpreter) { interpreter =>
-    interpreter.run()
-  }
+  scenarioInterpreter.run()
+  
+  Runtime.getRuntime.addShutdownHook(new Thread() {
```

Comment:
I understand that it will be invoked on SIGINT. Will be invoked after `System.exit` (e.g. by `UncaughtExceptionHandler`)? Why app was terminated immediately? By "immediately" you mean without logs?

Reply:
Yes, it will be also invoked on `System.exit`, I wonder if I should remove shutdownHook if we exit via UncaughtExceptionHandler...
The app was terminated immediately because `run` was in `Using.resource`, so `close` was invoked and we just gracefully finished :)